evalml/automl/automl_algorithm/default_algorithm.py

```diff
@@ -314,22 +356,36 @@ def next_batch(self):
         Returns:
             list(PipelineBase): a list of instances of PipelineBase subclasses, ready to be trained and evaluated.
         """
-        if self._batch_number == 0:
-            next_batch = self._create_naive_pipelines()
-        elif self._batch_number == 1:
-            next_batch = self._create_naive_pipelines(use_features=True)
-        elif self._batch_number == 2:
-            next_batch = self._create_fast_final()
-        elif self.batch_number == 3:
-            next_batch = self._create_ensemble()
-        elif self.batch_number == 4:
-            next_batch = self._create_long_exploration(n=self.top_n)
-        elif self.batch_number % 2 != 0:
-            next_batch = self._create_ensemble()
+        if self._ensembling:
+            if self._batch_number == 0:
+                next_batch = self._create_naive_pipelines()
+            elif self._batch_number == 1:
+                next_batch = self._create_naive_pipelines(use_features=True)
+            elif self._batch_number == 2:
+                next_batch = self._create_fast_final()
+            elif self.batch_number == 3:
+                next_batch = self._create_ensemble()
+            elif self.batch_number == 4:
+                next_batch = self._create_long_exploration(n=self.top_n)
+            elif self.batch_number % 2 != 0:
+                next_batch = self._create_ensemble()
+            else:
+                next_batch = self._create_n_pipelines(
+                    self._top_n_pipelines, self.num_long_pipelines_per_batch
+                )
         else:
-            next_batch = self._create_n_pipelines(
-                self._top_n_pipelines, self.num_long_pipelines_per_batch
-            )
+            if self._batch_number == 0:
```

Comment:
This means `DefaultAlgorithm` will have different batches for TS and non-TS problems as ensembling is turned off for TS.  Moreover, the pipeline structures are different for TS problems as well (we defer pipeline creation to `_make_pipeline_time_series` which incorporates the KIA work @freddyaboulton did). For the future but this begs the question: should we have a separate automl algorithm implementation for TS problems vs non-TS problems? I can envision the batch structure being the same (once ensembling for TS is fixed) but with a different implementation (pipelines are different etc.).

Reply:
If the mechanics of this algorithm for non-time series pipelines yield good results when applied to time series pipelines without violating any "rules of machine learning", e.g. data-leakage etc, then we should use the same algorithm.

I suspect that will be the case when we get ensembling to work for time series.